### DISCUSSION TITLE
Displaying Real-Time Power Consumption on KlipperScreen

### DISCUSSION THREAD

**Discussion by BenJule** [KlipperScreen/KlipperScreen#1518] [Q&A]
Dear KlipperScreen community, is it possible to display the current power consumption, similar to the chart for the nozzle, bed, and temperature?

I at least have an approach for the InfluxDB connection:

```
from influxdb_client import InfluxDBClient
import time

url = { URL }
token = { TOKEN }
org = { ORG }
bucket = { BUCKET }

client = InfluxDBClient(url=url, token=token, org=org)
query = f'from(bucket: "{bucket}") |> range(start: -5m)'

while True:
    tables = client.query_api().query(query)

    for table in tables:
        for record in table.records:
            temp = record.get_value()  # Temperaturwert
            power = record.get_value()  # Stromverbrauchswert

            with open('/home/pi/klipper/display_stats.txt', 'w') as f:
                f.write(f"Temperature: {temp} °C\n")
                f.write(f"Power: {power} W\n")

    time.sleep(60)
```

![Screenshot_20241129_113911](https://github.com/user-attachments/assets/9b0c67ef-c6dc-44e5-b4e4-ab0874a7d0cc)
![Screenshot_20241129_114425](https://github.com/user-attachments/assets/ac057486-9b99-4c6e-8beb-86243896a1a2)
![Screenshot_20241129_114448](https://github.com/user-attachments/assets/2a8d4eb8-1199-439e-bbf8-5ad08f876ba5)

**Comment by alfrix** [comment]
you can show power (duty cycle only) it's in the settings
